x/programs/cmd/simulator/src/lib.rs

```diff
@@ -169,10 +197,27 @@ pub struct PlanResult {
     /// The timestamp of the function call response.
     pub timestamp: u64,
     /// The result of the function call.
-    #[serde_as(as = "Base64")]
+    #[serde(deserialize_with = "b64_decode")]
     pub response: Vec<u8>,
 }
 
+fn b64_encode<S>(text: &String, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
```suggestion
fn b64_encode<S>(text: &str, serializer: S) -> Result<S::Ok, S::Error>
```

Reply:
And while you're at it... I really would prefer:
```suggestion
fn base64_encode<S>(text: &str, serializer: S) -> Result<S::Ok, S::Error>
```
But this is a nit

Reply:
Fixed!